src/analyze/State.re

```diff
@@ -266,34 +268,63 @@ let newBsPackage = (~reportDiagnostics, state, rootPath) => {
 };
 
 let newJbuilderPackage = (~reportDiagnostics, state, rootPath) => {
-  let rec findJbuilderProjectRoot = (path, buildDir) => {
+  let rec findJbuilderProjectRoot = (path) => {
```

Comment:
maybe my work in https://github.com/jaredly/reason-language-server/pull/174 will avoid changing this function?

Reply:
I'm not sure if you had something specific in mind, but I don't see anything in #174 that would obviate these changes. This has to run early, when all we have is the path to an opened source file, and needs to traverse upward until we find a project root dir, which is marked by the presence of any of the known dir names listed in here. Depending on a `buildDir` arg that comes from invoking `esy` would presume we've already detected it's an `esy`-managed project, and in the case of an opam-managed project I'd rather not have to invoke `esy` at all if it can be avoided.

Reply:
Sounds good!